projects/storefrontlib/shared/components/media/media.component.ts

```diff
@@ -91,7 +97,7 @@ export class MediaComponent implements OnChanges {
    */
   protected create(): void {
     this.media = this.mediaService.getMedia(
-      this.container,
+      this.container instanceof Array ? this.container[0] : this.container,
```

Comment:
The `container` should never been an array. We make `container` as `ImageGroup | ImageGroup[]`, because product's `PRIMARY` is `ImageGroup` and `GALLERY` is `ImageGroup[]`. We don't separate `PRIMARY` or `GALLERY`. But, there is no use case, we pass `GALLERY` list to it. (We can pass one item in `GALLAERY` to it). So, if the list is passed (should never happen), then we only process the first one.